models/glif_psc.h

```diff
@@ -0,0 +1,354 @@
+    // voltage-dependent component of the threshold in 1/ms
+    std::vector< double > asc_init_; // initial values of ASCurrents_ in pA
+    std::vector< double > k_;        // predefined time scale in 1/ms
+    std::vector< double > asc_amps_; // in pA
+    std::vector< double > r_;        // coefficient
+    std::vector< double > tau_syn_;  // synaptic port time constants in ms
+
+    // boolean flag which indicates whether the neuron has connections
+    bool has_connections_;
+
+    model_type glif_model_;
+
+    size_t n_receptors_() const; //!< Returns the size of tau_syn_
+
+    Parameters_();
+
+    void get( DictionaryDatum& ) const;
+    double set( const DictionaryDatum& );
+  };
+
+
+  struct State_
+  {
+    double U_;                         // relative membrane potential in mV
+    std::vector< double > ASCurrents_; // after-spike currents in pA
+    double ASCurrents_sum_;            // in pA
+    double threshold_;                 // voltage threshold in mV
+
+    double I_;                 // external current in pA
+    double I_syn_;             // post synaptic current in pA
+    std::vector< double > y1_; // synapse current evolution state 1 in pA
+    std::vector< double > y2_; // synapse current evolution state 2 in pA
+
+    State_( const Parameters_& );
+
+    void get( DictionaryDatum&, const Parameters_& ) const;
+    void set( const DictionaryDatum&, const Parameters_&, double );
+  };
+
+
+  struct Buffers_
+  {
+    Buffers_( glif_psc& );
+    Buffers_( const Buffers_&, glif_psc& );
+
+    std::vector< nest::RingBuffer > spikes_; //!< Buffer incoming spikes through delay, as sum
+    nest::RingBuffer currents_;              //!< Buffer incoming currents through delay,
+
+    //! Logger for all analog data
+    nest::UniversalDataLogger< glif_psc > logger_;
+  };
+
+  struct Variables_
+  {
+    double t_ref_remaining_; // counter during refractory period, in ms
+    double t_ref_total_;     // total time of refractory period, in ms
+    double last_spike_;      // threshold spike component in mV
+    double last_voltage_;    // threshold voltage component in mV
+
+    std::vector< double > P11_; // synaptic current evolution parameter
+    std::vector< double > P21_; // synaptic current evolution parameter
+    std::vector< double > P22_; // synaptic current evolution parameter
+    double P30_;                // membrane current/voltage evolution parameter
+    double P33_;                // membrane voltage evolution parameter
+    std::vector< double > P31_; // synaptic/membrane current evolution parameter
+    std::vector< double > P32_; // synaptic/membrane current evolution parameter
+
+    /** Amplitude of the synaptic current.
+              This value is chosen such that a post-synaptic current with
+              weight one has an amplitude of 1 pA.
+    */
+    std::vector< double > PSCInitialValues_;
+
+    unsigned int receptor_types_size_;
```

Comment:
Wouldn't `num_receptors_` or `num_synapses_` be a better name?

Reply:
It is a redundant variable, has been removed.